glusterapi/volume.py

```diff
@@ -1,132 +1,275 @@
+
+        voloptions = dict()
+        req = dict()
+        for key in options:
+            voloptions[key] = options[key]
+        req['options'] = voloptions
+        req['advanced'] = advance
+        req['experimental'] = experimental
+        req['deprecated'] = deprecated
+        return self._handle_request(self._post, httplib.OK,
+                                    "/v1/volumes/%s/options" % vol_name,
+                                    json.dumps(req))
 
-    def volume_reset(self, volname, optnames=[]):
+    def volume_reset(self, vol_name, options=None):
         """
-        Start Gluster Volume
+        Start Gluster Volume.
 
         :param volname: (string) Volume Name
-        :param force: (bool) Stop Volume with Force
+        :param options: (dict) options to set on volume
         :raises: GlusterAPIError or failure
         """
+        vol_name = vol_name.strip()
+        if not vol_name:
+            raise GlusterApiInvalidInputs("Volume name cannot be empty")
+
         pass
 
-    def volume_info(self, volname=None):
+    def volume_info(self, vol_name):
         """
-        Gluster Volume Info
+        Gluster Volume Info.
 
         :param volname: (string) Volume Name
-        :param force: (bool) Stop Volume with Force
-        :raises: GlusterAPIError or failure
+        :raises: GlusterAPIError on failure
         """
-        pass
+        vol_name = vol_name.strip()
+        if not vol_name:
+            raise GlusterApiInvalidInputs("Volume name cannot be empty")
 
-    def volume_status(self, volname=None):
+        return self._handle_request(self._get, httplib.OK,
+                                    '/v1/volumes/%s/bricks' % vol_name)
+
+    def volume_status(self, vol_name=None):
         """
-        Gluster Volume Status
+        Gluster Volume Status.
 
-        :param volname: (string) Volume Name
-        :raises: GlusterAPIError or failure
+        :param vol_name: (string) Volume Name
+        :raises: GlusterAPIError on failure
         """
-        pass
+        vol_name = vol_name.strip()
+        if not vol_name:
+            raise GlusterApiInvalidInputs("Volume name cannot be empty")
+
+        return self._handle_request(self._get, httplib.OK,
+                                    '/v1/volumes/%s/status' % vol_name)
 
-    def volume_get(self, volname, optnames=[]):
+    def volume_get(self, vol_name, options=None):
         """
-        Get Gluster Volume Options
+        Get Gluster Volume Options.
 
-        :param volname: (string) Volume Name
-        :param force: (bool) Stop Volume with Force
-        :raises: GlusterAPIError or failure
+        :param vol_name: (string) Volume Name
+        :param options: (dict) get volumes based on options
+        :raises: GlusterAPIError on failure
         """
+        vol_name = vol_name.strip()
+        if not vol_name:
+            raise GlusterApiInvalidInputs("Volume name cannot be empty")
```

Comment:
Maybe you can leave a comment here with a TODO that you'll complete this once the PR is merged to GD2.